### DISCUSSION TITLE
sending data to an external html page to display via the external_html plugin

### DISCUSSION THREAD

**Discussion by Quinlanptq** [jspsych/jsPsych#1002] [Q&A] [answered]
I am trying to update a variable on an external html page. I am using the external_html plugin. It is a simple memory span task so after seeing a sequence of items they are presented with a sequence of text boxes one after the other.
I want the page to display "Input Item number 1" as a label.

I have managed to get this to work if I simply call the associated html page using

recall1.html?pqname=1
where the value of pqname increments over the sequence and is displayed correctly in the label on the page.

So the link in the browser works fine but if I try passing the full url to the external_html plugin in my JSpsych code it displays

Input Item number undefined

I cant figure out what the problem is and how to fix it. Any help gratefully received.

On an ancillary note my recall2.html page has JS code running in it that I can easily step through in the developer tools when I call it directly in the browser but I am at loss how to debug this when I run my JSpsch code - any pointers how to do this?

Thanks in anticipation
Philip

Philip Quinlan
Psychology, York, UK

**Comment by jodeleeuw** [comment, marked as answer]
Hi Philip,

Confession: the external-html plugin was built for a fairly limited set of use cases. Mostly around having a simple form for people to fill out, like informed consent. I don't know how well it will handle URL variables. I suspect that might be beyond the scope of how the plugin is configured.

It sounds like you are using the external page for a text box entry. Did you consider using the `survey-text` plugin?

**Comment by Quinlanptq** [comment]
Thanks very much for your response Josh.
I took a look at the plugin and it wasn't quite what I wanted.
I ditched the idea of passing the loop counter via the url and decided to use localStorage instead.
My declaration of the external-html looks like this

var recall_it = {
            type:'external-html',
            url: big_url,
            execute_script: true,
            cont_key: 13,
            check_fn: check_item,
            on_start: function (){
                  var this_label = "Input Item number ".concat(kounter.toString());
                  localStorage.setItem("lable",this_label);
                  kounter++;
                  if (kounter>7) {
                      kounter=1;
                  }....

kounter is a 'global' variable 

then I can pick up this value in the external html page by 

using this

        $(document).ready(function(){
           
            var x = localStorage.getItem("lable");

           $("#name").text(x);

This seems to work fine.

So I can have a labeled text box on a new page that allows the participant to type in a response and by pressing RETURN it will move onto the next page and so on.

In this way they can just type and not have to bother with pressing a submit key after every response.

Thanks again for a really great web resource
Best
Philip.

**Reply by jodeleeuw** [reply]
You'll need to attach an event listener to the `input` event of the text box, and end the trial based on that event.

It sounds like you're rapidly approaching *make a new plugin* territory. If you've gone through the trouble of setting up an external html page, converting that logic to a plugin should be relatively straight forward.

I have a tutorial video on creating plugins that might be useful for the conceptual side of it: https://www.youtube.com/watch?v=XQcsFwAmbiw
